tests/lib/rules/require-default-props.js

```diff
@@ -1896,6 +1910,102 @@ ruleTester.run('require-default-props', rule, {
       errors: [{
         message: 'propType "first-name" is not required, but has no corresponding defaultProp declaration.'
       }]
+    },
+    {
+      code: [
+        'class Hello extends React.Component {',
+        '  render() {',
+        '    return <div>Hello {this.props.foo}</div>;',
+        '  }',
+        '}',
+        'Hello.propTypes = {',
+        '  foo: PropTypes.string.isRequired',
+        '};',
+        'Hello.defaultProps = {',
+        '  foo: \'bar\'',
+        '};'
+      ].join('\n'),
+      options: [{forbidDefaultForRequired: true}],
+      errors: [{
+        message: 'propType "foo" is required and should not have a defaultProp declaration.'
+      }]
+    },
+    {
+      code: [
+        'function Hello(props) {',
+        '  return <div>Hello {props.foo}</div>;',
+        '}',
+        'Hello.propTypes = {',
+        '  foo: PropTypes.string.isRequired',
+        '};',
+        'Hello.defaultProps = {',
+        '  foo: \'bar\'',
+        '};'
+      ].join('\n'),
+      options: [{forbidDefaultForRequired: true}],
+      errors: [{
+        message: 'propType "foo" is required and should not have a defaultProp declaration.'
+      }]
+    },
+    {
+      code: [
+        'const Hello = (props) => {',
+        '  return <div>Hello {props.foo}</div>;',
+        '};',
+        'Hello.propTypes = {',
+        '  foo: PropTypes.string.isRequired',
+        '};',
+        'Hello.defaultProps = {',
+        '  foo: \'bar\'',
+        '};'
+      ].join('\n'),
+      options: [{forbidDefaultForRequired: true}],
+      errors: [{
+        message: 'propType "foo" is required and should not have a defaultProp declaration.'
+      }]
+    },
+    {
+      code: [
+        'class Hello extends React.Component {',
+        '  static propTypes = {',
+        '    foo: PropTypes.string.isRequired',
+        '  }',
+        '  static defaultProps = {',
+        '    foo: \'bar\'',
+        '  }',
+        '  render() {',
+        '    return <div>Hello {this.props.foo}</div>;',
+        '  }',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint',
+      options: [{forbidDefaultForRequired: true}],
+      errors: [{
+        message: 'propType "foo" is required and should not have a defaultProp declaration.'
+      }]
+    },
+    {
+      code: [
+        'class Hello extends React.Component {',
+        '  static get propTypes () {',
+        '    return {',
+        '      foo: PropTypes.string.isRequired',
+        '    };',
+        '  }',
+        '  static get defaultProps() {',
+        '    return {',
+        '      foo: \'bar\'',
+        '    };',
+        '  }',
+        '  render() {',
+        '    return <div>Hello {this.props.foo}</div>;',
+        '  }',
+        '}'
+      ].join('\n'),
+      options: [{forbidDefaultForRequired: true}],
+      errors: [{
+        message: 'propType "foo" is required and should not have a defaultProp declaration.'
```

Comment:
Will this also warn on the case where a spread property has an object literal referenced in the same file? 
eg: 
```js
const defaults = {
  foo: ''
}

class Hello extends React.Component {
  static propTypes = {
    foo: PropTypes.string.isRequired
  }

  static defaultProps = { ...defaults } // <---

  render () {
    return <div>Hello {this.props.foo}</div>;
  }
}
```

Reply:
nvm read your pr discription